packages/expo-in-app-purchases/android/src/main/java/expo/modules/inapppurchases/BillingManager.java

```diff
@@ -253,27 +259,39 @@ public int getBillingClientResponseCode() {
   }
 
   /**
-   * Query purchases across various use cases and deliver the result in a formalized way through
-   * a listener
+   * Query both in app purchases and subscriptions and deliver the result in a formalized way 
+   * through a listener
    */
   public void queryPurchases(final Promise promise) {
     Runnable queryToExecute = new Runnable() {
+      BillingResult inAppBillingResult;
+      List<Purchase> purchases = new ArrayList<>();
```

Comment:
Move these declarations inside of `run()`. They might need to be marked `final`.